src/loader/lib/transpiler.js

```diff
@@ -2,121 +2,163 @@
+				}
+
+				return (transpiler.Compiler ? traceurTranspile : babelTranspile)
+					.call(self, load, transpiler);
+			})
+			.then(function(code) {
+				return 'var __moduleAddress = "' + load.address + '";' + code;
+			});
+	};
+
+	Loader.prototype.instantiate = function(load) {
+		var self = this;
+		return Promise.resolve(self.normalize(self.transpiler))
+			.then(function(transpilerNormalized) {
+				// load transpiler as a global (avoiding System clobbering)
+				if (load.name === transpilerNormalized) {
+					return {
+						deps: [],
+						execute: function() {
+							var curSystem = g.System;
+							var curLoader = g.Reflect.Loader;
+							// ensure not detected as CommonJS
+							__eval('(function(require,exports,module){' + load.source + '})();', g, load);
+							g.System = curSystem;
+							g.Reflect.Loader = curLoader;
+							return getTranspilerModule(self, load.name);
+						}
+					};
+				}
+			});
+	};
+
+	function traceurTranspile(load, traceur) {
+		var options = this.traceurOptions || {};
+		options.modules = 'instantiate';
+		options.script = false;
+		options.sourceMaps = 'inline';
+		options.filename = load.address;
+		options.inputSourceMap = load.metadata.sourceMap;
+		options.moduleName = false;
+
+		var compiler = new traceur.Compiler(options);
+		var source = doTraceurCompile(load.source, compiler, options.filename);
+
+		// add "!eval" to end of Traceur sourceURL
+		// I believe this does something?
+		source += '!eval';
+
+		return source;
+	}
+	function doTraceurCompile(source, compiler, filename) {
+		try {
+			return compiler.compile(source, filename);
+		}
+		catch(e) {
+			// traceur throws an error array
+			throw e[0];
+		}
+	}
+
+	function registerCustomPlugins(babel, plugins) {
+		var promises = [];
+		var self = this;
+
+		function isAvailable(plugin) {
+			var availablePlugins = babel.availablePlugins || {};
+
+			return !!availablePlugins[plugin];
+		}
+
+		plugins.forEach(function(plugin) {
+			if (!isAvailable(plugin)) {
+				promises.push(self.import(plugin, { name: self.configMain })
+					.then(function(module) {
+						babel.registerPlugin(plugin, module);
+					}));
+			}
+		});
+
+		return Promise.all(promises);
+	}
+
+	function getBabelOptions(load, babel) {
+		var options = this.babelOptions || {};
+
+		options.sourceMap = 'inline';
+		options.filename = load.address;
+		options.code = true;
+		options.ast = false;
+
+		var babelVersion = babel.version ? +babel.version.split(".")[0] : 6;
+		if (!babelVersion) babelVersion = 6;
+
+		if (babelVersion >= 6) {
+			// delete the old babel options if they are present in config
+			delete options.optional;
+			delete options.whitelist;
+			delete options.blacklist;
+
+			options.presets = options.presets || [];
+			options.plugins = options.plugins || [];
+
+			// make sure presents and plugins needed for Steal to work
+			// correctly are set
+			options.presets.unshift("es2015-no-commonjs", "react", "stage-0");
```

Comment:
this has to happen only once, otherwise we'll unshift the default for each module loaded.